kafka/src/main/java/com/bazaarvoice/emodb/kafka/metrics/DropwizardMetricsReporter.java

```diff
@@ -0,0 +1,87 @@
+package com.bazaarvoice.emodb.kafka.metrics;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.SharedMetricRegistries;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.metrics.KafkaMetric;
+import org.apache.kafka.common.metrics.MetricsReporter;
+import org.coursera.metrics.datadog.TaggedName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DropwizardMetricsReporter implements MetricsReporter {
+
+    private static Logger _log = LoggerFactory.getLogger(DropwizardMetricsReporter.class);
+
+    private static final String REGISTRY_NAME = "default";
+    private static final String METRIC_PREFIX = MetricsReporter.class.getPackage().getName();
+
+    protected MetricRegistry _registry;
+    private Set<String> _metricNames = new HashSet<>();
+
+    public static void registerDefaultMetricsRegistry(MetricRegistry metricRegistry) {
+        SharedMetricRegistries.add(REGISTRY_NAME, metricRegistry);
+    }
+
+    @Override
+    public void configure(Map<String, ?> configs) {
+    }
+
+    @Override
+    public void init(List<KafkaMetric> metrics) {
+        _registry = SharedMetricRegistries.getOrCreate(REGISTRY_NAME);
+        for (KafkaMetric kafkaMetric : metrics) {
+            metricChange(kafkaMetric);
+        }
+    }
+
+    @Override
+    public void metricChange(KafkaMetric metric) {
+        _log.debug("Processing a metric change for {}", metric.metricName().toString());
+        String name = metricName(metric);
+
+        Gauge<Double> gauge = metric::value;
```

Comment:
I'm ok with changing this very soon in a separate PR, but I think changing it is going to require extra testing for compatibility because `metricValue()` return `Object` and not `double`